### PR TITLE
Display selected students, not selected student, and companies

### PR DESCRIPTION
**Pull request by hasinichathu**

### CODE REVIEW

File: Frontend/src/app/Models/basic-company.model.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export class BasicCompany {
+    userName: string;
+    companyName:string;
+    phoneNumber: number;
+    email:string;
+    location:string;
+    numberOfvacancies:number;
+    filledVacancies: number;
+    Website: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-company.model.ts:9]:
```suggestion
    website: string;
```

File: Frontend/src/app/Models/basic-company.model.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    contactPersonName: string;
+    contactPersonphoneNumber: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-company.model.ts:11]:
```suggestion
    contactPersonPhoneNumber: string;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export class BasicStudent {
+    IndexNumber: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:2]:
```suggestion
    indexNumber: string;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    FullName: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:3]:
```suggestion
    fullName: string;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    NameWithInitials: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:4]:
```suggestion
    nameWithInitials: string;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    PhoneNumber : number;
+    Email: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:6]:
```suggestion
    email: string;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    Sem1GPA : number;
+    Sem2GP : number;
+    Sem3GPA : number;
+    Sem4GPA : number;
+    SGPA : number;
+    BatchId: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:12]:
```suggestion
    batchId: string;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    IsSelected: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:13]:
```suggestion
    isSelected: string;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    CompanyId: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:14]:
```suggestion
    companyId: string;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    CompanyName: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:15]:
```suggestion
    companyName: string;
```

File: Frontend/src/app/Services/feedback.service.spec.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
```
Comment (yashodgayashan) [inline Frontend/src/app/Services/feedback.service.spec.ts:2]: Do we need a space here?

File: Frontend/src/app/Services/admin.service.spec.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
```
Comment (yashodgayashan) [inline Frontend/src/app/Services/admin.service.spec.ts:2]: Do we need a space here?

File: Frontend/src/app/Services/company.service.ts [hunk 13]
```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs/internal/Observable';
+import { BasicCompany} from '../Models/basic-company.model';
+import { environment } from '../../environments/environment'
```
Comment (yashodgayashan) [inline Frontend/src/app/Services/company.service.ts:5]:
```suggestion
import { environment } from '../../environments/environment';
```

File: Frontend/src/app/Services/student.service.ts [hunk 14]
```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs/internal/Observable';
+import { Student} from 'src/app/layouts/admin-layout/models/student.model';
+import { environment } from '../../environments/environment'
```
Comment (yashodgayashan) [inline Frontend/src/app/Services/student.service.ts:5]:
```suggestion
import { environment } from '../../environments/environment';

```

File: Frontend/src/app/layouts/Models/basic-company.model.ts [hunk 15]
```diff
@@ -0,0 +1,14 @@
+export class BasicCompany {
+    userName: string;
+    companyName:string;
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/Models/basic-company.model.ts:3]:
```suggestion
    companyName: string;
```

File: Frontend/src/app/layouts/Models/basic-company.model.ts [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+    phoneNumber: number;
+    email:string;
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/Models/basic-company.model.ts:5]:
```suggestion
    email: string;
```

File: Frontend/src/app/layouts/Models/basic-company.model.ts [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+    location:string;
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/Models/basic-company.model.ts:6]:
```suggestion
    location: string;
```

File: Frontend/src/app/layouts/Models/basic-company.model.ts [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+    numberOfvacancies:number;
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/Models/basic-company.model.ts:7]:
```suggestion
    numberOfVacancies: number;
```

File: Frontend/src/app/layouts/Models/basic-company.model.ts [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+    filledVacancies: number;
+    Website: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/Models/basic-company.model.ts:9]:
```suggestion
    website: string;
```

File: Frontend/src/app/layouts/Models/basic-company.model.ts [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+    contactPersonName: string;
+    contactPersonphoneNumber: string;
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/Models/basic-company.model.ts:11]:
```suggestion
    contactPersonPhoneNumber: string;
```

File: Frontend/src/app/layouts/Models/basic-company.model.ts [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+    contactPersonemail:string;
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/Models/basic-company.model.ts:12]:
```suggestion
    contactPersonEmail: string;
```

File: Frontend/src/app/Models/basic-company.model.ts [hunk 22]
```diff
@@ -0,0 +1,13 @@
+export class BasicCompany {
+    userName: string;
+    companyName:string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-company.model.ts:3]:
```suggestion
    companyName: string;
```

File: Frontend/src/app/Models/basic-company.model.ts [hunk 23]
```diff
@@ -0,0 +1,13 @@
...
+    phoneNumber: number;
+    email:string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-company.model.ts:5]:
```suggestion
    email: string;
```

File: Frontend/src/app/Models/basic-company.model.ts [hunk 24]
```diff
@@ -0,0 +1,13 @@
...
+    location:string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-company.model.ts:6]:
```suggestion
    location: string;
```

File: Frontend/src/app/Models/basic-company.model.ts [hunk 25]
```diff
@@ -0,0 +1,13 @@
...
+    numberOfvacancies:number;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-company.model.ts:7]:
```suggestion
    numberOfVacancies: number;
```

File: Frontend/src/app/Models/basic-company.model.ts [hunk 26]
```diff
@@ -0,0 +1,13 @@
...
+    filledVacancies: number;
+    Website: string;
+    contactPersonName: string;
+    contactPersonphoneNumber: string;
+    contactPersonemail:string;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-company.model.ts:12]:
```suggestion
    contactPersonEmail: string;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 27]
```diff
@@ -0,0 +1,16 @@
+export class BasicStudent {
+    IndexNumber: string;
+    FullName: string;
+    NameWithInitials: string;
+    PhoneNumber : number;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:5]:
```suggestion
    phoneNumber: number;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 28]
```diff
@@ -0,0 +1,16 @@
...
+    Email: string;
+    Sem1GPA : number;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:7]:
```suggestion
    sem1GPA: number;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 29]
```diff
@@ -0,0 +1,16 @@
...
+    Sem2GP : number;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:8]:
```suggestion
    sem2GPA: number;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 30]
```diff
@@ -0,0 +1,16 @@
...
+    Sem3GPA : number;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:9]:
```suggestion
    sem3GPA: number;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 31]
```diff
@@ -0,0 +1,16 @@
...
+    Sem4GPA : number;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:10]:
```suggestion
    sem4GPA: number;
```

File: Frontend/src/app/Models/basic-student.model.ts [hunk 32]
```diff
@@ -0,0 +1,16 @@
...
+    SGPA : number;
```
Comment (yashodgayashan) [inline Frontend/src/app/Models/basic-student.model.ts:11]:
```suggestion
    SGPA: number;
```

File: Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html [hunk 33]
```diff
@@ -51,270 +51,33 @@
...
-                  Portugal
-                </td>
-                <td>
-                  Gloucester
-                </td>
-                <td class="text-right">
-                  $98,615
-                </td>
+              <tr *ngfor ="let student of students">
+                <td> {{student.IndexNumber}}</td>
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html:74]:
```suggestion
                <td> {{student.indexNumber}}</td>
```

File: Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html [hunk 34]
```diff
@@ -51,270 +51,33 @@
...
+                <td> {{student.NameWithInitials}}</td>
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html:75]:
```suggestion
                <td> {{student.nameWithInitials}}</td>
```

File: Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html [hunk 35]
```diff
@@ -51,270 +51,33 @@
...
+                <td> {{student.PhoneNumber}}</td>
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html:76]:
```suggestion
                <td> {{student.phoneNumber}}</td>
```

File: Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html [hunk 36]
```diff
@@ -51,270 +51,33 @@
...
+                <td> {{student.SGPA}}</td>
+                <td> {{student.BatchId}}</td>
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html:78]:
```suggestion
                <td> {{student.batchId}}</td>
```

File: Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html [hunk 37]
```diff
@@ -51,270 +51,33 @@
...
+                <td> {{student.CompanyName}}</td>
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html:79]:
```suggestion
                <td> {{student.companyName}}</td>
```

File: Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html [hunk 38]
```diff
@@ -51,270 +51,33 @@
...
+                <td> {{student.Email}}</td>
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.html:80]:
```suggestion
                <td> {{student.email}}</td>
```

File: Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.ts [hunk 39]
```diff
@@ -1,15 +1,20 @@
 import { Component, OnInit } from '@angular/core';
-
+import { Student } from '../../models/student.model';
+import { StudentService } from '../../../../Services/student.service';
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.ts:3]:
```suggestion
import { StudentService } from '../../../../Services/student.service';

```

File: Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.ts [hunk 40]
```diff
@@ -1,15 +1,20 @@
...
 @Component({
   selector: 'app-not-selected-students',
   templateUrl: './not-selected-students.component.html',
   styleUrls: ['./not-selected-students.component.scss']
 })
 export class NotSelectedStudentsComponent implements OnInit {
-
-  constructor() { }
+  students : Student[];
+  batchId : String = "batch16";
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.ts:11]: I think we need an API to get the latest batch or ?

File: Frontend/src/app/layouts/admin-layout/pages/organizations/organizations.component.html [hunk 41]
```diff
@@ -12,113 +12,33 @@ <h4 class="card-title"> Simple Table</h4>
...
-                <td>
-                  Gloucester
-                </td>
-                <td class="text-right">
-                  $98,615
-                </td>
+              <tr *ngFor ='let company of companies'>
+                <td>{{company.companyName}}</td>
+                <td>{{company.location}}</td>
+                <td>{{company.numberOfvacancies}}</td>
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/organizations/organizations.component.html:37]:
```suggestion
                <td>{{company.numberOfVacancies}}</td>
```

File: Frontend/src/app/layouts/admin-layout/pages/organizations/organizations.component.html [hunk 42]
```diff
@@ -12,113 +12,33 @@ <h4 class="card-title"> Simple Table</h4>
...
+                <td>{{company.filledVacancies}}</td>
+                <td>{{company.contactPersonName}}</td>
+                <td>{{company.contactPersonphoneNumber}}</td>
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/organizations/organizations.component.html:40]:
```suggestion
                <td>{{company.contactPersonPhoneNumber}}</td>
```

File: Frontend/src/app/layouts/admin-layout/pages/organizations/organizations.component.html [hunk 43]
```diff
@@ -12,113 +12,33 @@ <h4 class="card-title"> Simple Table</h4>
...
+                <td>{{company.contactPersonemail}}</td>
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/organizations/organizations.component.html:41]:
```suggestion
                <td>{{company.contactPersonEmail}}</td>
```

File: Frontend/src/app/layouts/admin-layout/pages/organizations/organizations.component.ts [hunk 44]
```diff
@@ -1,15 +1,22 @@
...
   selector: 'app-organizations',
   templateUrl: './organizations.component.html',
   styleUrls: ['./organizations.component.scss']
 })
 export class OrganizationsComponent implements OnInit {
-
-  constructor() { }
+  companies:BasicCompany[];
+  constructor(private comService: CompanyService) {
+
```
Comment (yashodgayashan) [inline Frontend/src/app/layouts/admin-layout/pages/organizations/organizations.component.ts:13]: Please remove these spaces

**Comment by yashodgayashan** [comment]
Folder structure is okay but please stick to a proper naming convention coz in `fronend>src>app` some folders are started with capital while most of are started with simple.

File: Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.ts (hunk 40 above)
Comment (hasinichathu) [reply Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.ts:11]: I thought to load batches to a dropdown. and  from there admin selects the batch she wants.

Comment (yashodgayashan) [reply Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.ts:11]: Ahhh thats fine. But what will be the initial value before selecting the dropdown?

Comment (hasinichathu) [reply Frontend/src/app/layouts/admin-layout/pages/not-selected-students/not-selected-students.component.ts:11]: It should be current batch

File: Frontend/src/app/Services/feedback.service.spec.ts (hunk 11 above)
Comment (hasinichathu) [reply Frontend/src/app/Services/feedback.service.spec.ts:2]: this is an auto generated import. If im not mistaken angular documentation shows their imports like this.

File: Frontend/src/app/Services/admin.service.spec.ts (hunk 12 above)
Comment (hasinichathu) [reply Frontend/src/app/Services/admin.service.spec.ts:2]: this is an auto generated import.

**Pull request merged**